csharp/ql/lib/semmle/code/csharp/commons/StructuralComparison.qll

```diff
@@ -111,18 +103,24 @@ private GvnKind getGvnKind(ControlFlowElement cfe) {
     result = TGvnKindDeclaration(kind, isTargetThis, d)
   )
   or
-  not exists(referenceAttribute(cfe)) and
-  result = getKind(cfe)
+  exists(int kind |
+    not exists(referenceAttribute(cfe)) and
```

Comment:
Ah yes, we can push that into the disjunction as this is only relevant for expressions